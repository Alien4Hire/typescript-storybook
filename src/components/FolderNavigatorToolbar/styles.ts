import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
