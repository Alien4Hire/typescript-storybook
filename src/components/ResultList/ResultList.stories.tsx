import { ComponentMeta, ComponentStory } from '@storybook/react';

import ProviderWrapper from '../../util/ProviderWrapper';
import Component, { ResultListPropsT } from './ResultList';

export default {
  title: 'Component/ResultList',
  component: Component,
  decorators: [
    (Story) => (
      <ProviderWrapper>
        <Story />
      </ProviderWrapper>
    ),
  ],
} as ComponentMeta<typeof Component>;

const Template: ComponentStory<typeof Component> = (args) => {
  return <Component {...args} />;
};

export const ResultList = Template.bind({});
ResultList.args = {
  result: [
    {
      checked: false,
      checkedValue: 'test',
      activityName: 'Ch 1 LearningCurve Adaptive Quiz',
      contentId: '1',
      modifiedDate: 'xx/xx/xxxxx at xx:xx',
      addedToLibrary: false,
    },
    {
      checked: false,
      checkedValue: 'test',
      activityName: 'Ch 1 LearningCurve Adaptive Quiz',
      contentId: '2',
      modifiedDate: 'xx/xx/xxxxx at xx:xx',
      addedToLibrary: true,
    },
    {
      checked: true,
      checkedValue: 'test',
      activityName: 'Ch 1 LearningCurve Adaptive Quiz',
      contentId: '3',
      modifiedDate: 'xx/xx/xxxxx at xx:xx',
      addedToLibrary: false,
    },
  ],
  onAddResource: () => {},
  itemsOnPage: 100,
} as ResultListPropsT;
