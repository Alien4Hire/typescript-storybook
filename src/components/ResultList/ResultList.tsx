import { useState } from 'react';
import { useAppDispatch, useAppSelector } from '../../app/hooks';
import {
  addResourceToLibrary,
  getSearchResults,
  ResultItemT,
  setAllChecked,
  setChecked,
} from '../../app/reducers/search/searchSlice';
import { setToastNotification } from '../../app/reducers/appSlice/appSlice';
import { theme } from '../../theme';
import ResultItem from '../ResultItem/ResultItem';
import {
  Container,
  ItemsContainer,
  SelectAllContainer,
  StyledTitle,
} from './ResultList.styles';
import { Checkbox, FormControlLabel } from '@mui/material';
import { generateId } from '../RecursiveTreeView/utils';
import { addFolderItem as actionAddFolderItem } from '../../app/reducers/lessons/lessonsSlice';
import { FolderT } from '../../util/folderTree';
import { MlDropdownMenuOptionPropsT } from '../DropdownMenu/DropdownMenu';
import NoFilterResult from '../NoFilterResult';

export type ResultListPropsT = {
  result: ResultItemT[];
  itemsOnPage?: number;
};

const ResultList = ({ result, itemsOnPage = 100 }: ResultListPropsT) => {
  const dispatch = useAppDispatch();
  const { indeterminateStatus, allCheckedStatus } =
    useAppSelector(getSearchResults);
  const [currentPage] = useState<number>(1);
  const currentElements = itemsOnPage * currentPage;
  const from = currentElements - itemsOnPage;
  const to = currentElements > result.length ? result.length : currentElements;
  const title = result.length ? `Showing ${from + 1}-${to} of ${result.length} results` : 'Showing 0 of 0 results';

  const checkHandler = (id: string) => {
    dispatch(setChecked(id));
  };

  const checkAllHandler = () => {
    dispatch(setAllChecked());
  };

  const handleAddResource = (
    id: string,
    folder: MlDropdownMenuOptionPropsT
  ) => {
    const idx = result.findIndex((el) => el.contentId === id);
    dispatch(addResourceToLibrary(id));
    const file = {
      id: generateId(),
      name: result[idx].activityName,
      type: 'activity' as FolderT['type'],
      parentId: folder.value,
    };
    dispatch(actionAddFolderItem(file));
    dispatch(
      setToastNotification({
        open: true,
        message: `"${result[idx].activityName}" added to "${folder.label}".`,
        type: 'success',
      })
    );
  };

  return (
    <Container>
      <StyledTitle>{title}</StyledTitle>
      {
        result.length ? (
        <Container>
          <SelectAllContainer>
            <FormControlLabel
             control={
              <Checkbox
                value={'SelectAll'}
                checked={allCheckedStatus}
                indeterminate={indeterminateStatus}
                onChange={checkAllHandler}
              />
            }
            sx={{ margin: '0px' }}
            label="Select all"
          />
          </SelectAllContainer>
          <ItemsContainer theme={theme}>
          {result.map((el) => (
            <ResultItem
              key={el.contentId}
              checkedValue={el.checkedValue}
              checked={el.checked}
              onCheck={checkHandler}
              activityName={el.activityName}
              contentId={el.contentId}
              modifiedDate={el.modifiedDate}
              addedToLibrary={el.addedToLibrary}
              onAddResource={handleAddResource}
            />
          ))}
          </ItemsContainer>
        </Container>          
        ): (
          <NoFilterResult />
        )
      }
    </Container>
  );
};

export default ResultList;
