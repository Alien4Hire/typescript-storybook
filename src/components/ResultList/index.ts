export { default } from './ResultList';

export const mockResult = {
  checked: false,
  checkedValue: 'test',
  activityName: 'LearningCurve Adaptive Quiz',
  modifiedDate: 'xx/xx/xxxxx at xx:xx',
  addedToLibrary: false,
} as const;

export const mockResultList = [...Array(10)].map((_, index) => ({
  ...mockResult,
  contentId: String(index),
  title: `Ch ${index + 1} ${mockResult.activityName}`,
}));
