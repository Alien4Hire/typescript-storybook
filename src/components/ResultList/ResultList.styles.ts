import { Box, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';

export const Container = styled(Box)(() => ({
  width: '100%',
}));

export const SelectAllContainer = styled(Box)(() => ({
  width: '100%',
  padding: '12px 20px',
}));

export const ItemsContainer = styled(Box)(({ theme, ...props }) => ({
  alignItems: 'center',
  backgroundColor: props.checked
    ? theme.palette.blue.veryLight
    : theme.palette.white.main,
}));

export const StyledTitle = styled(Typography)({
  fontStyle: 'italic',
  fontSize: 15,
  lineHeight: '24px',
  marginBottom: '5px',
});
