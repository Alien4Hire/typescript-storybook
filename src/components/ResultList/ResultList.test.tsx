// import { fireEvent } from '@storybook/testing-library';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { store } from '../../app/store';
import '../../setupTests';
import { theme } from '../../theme';
import ResultList, { ResultListPropsT } from './ResultList';

const baseProps: ResultListPropsT = {
  result: [
    {
      checkedValue: 'mock-checked-value',
      checked: false,
      activityName: 'mock-activity-name',
      contentId: 'mock-content-id',
      modifiedDate: 'xx/xx/xxxx at xx/xx',
      addedToLibrary: false,
    },
  ],
  itemsOnPage: 1,
};

const Component = (props: ResultListPropsT) => (
  <Provider store={store}>
    <Router>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>
          <ResultList {...props} />
        </ThemeProvider>
      </StyledEngineProvider>
    </Router>
  </Provider>
);

const renderComponent = (customProps: Partial<ResultListPropsT> = {}) => {
  const props = { ...baseProps, ...customProps };

  return render(<Component {...props} />);
};

describe('<ResultItem />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it('should render the search result', () => {
    renderComponent();
    expect(screen.getByText(/mock-activity-name/)).toBeInTheDocument();
  });

  it('Should display the `Add Resource` button if resource is not added to library', () => {
    renderComponent();
    const addResourceText = screen.getByText(/Add Resource/);
    expect(addResourceText).toBeInTheDocument();
  });

  it('Should display `In Library` text if resource is added to library', () => {
    const customProps: ResultListPropsT = {
      result: [
        {
          checkedValue: 'mock-checked-value',
          checked: false,
          activityName: 'mock-activity-name',
          contentId: 'mock-content-id',
          modifiedDate: 'xx/xx/xxxx at xx/xx',
          addedToLibrary: true,
        },
      ],
    };
    renderComponent(customProps);
    const inLibraryText = screen.getByText(/In Library/);
    expect(inLibraryText).toBeInTheDocument();
  });
});
