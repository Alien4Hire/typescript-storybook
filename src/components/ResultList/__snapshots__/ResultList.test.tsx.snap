// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c5::-moz-focus-inner {
  border-style: none;
}

.c5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
}

.c15 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #00758e;
  margin-bottom: 7px;
}

.c16 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #383838;
}

.c19 {
  margin: 0;
  color: #080808;
  font-size: 13px;
  font-family: Source Sans Pro;
  font-weight: 400;
  font-style: normal;
  color: #383838;
  margin-top: 3px;
  line-height: 15px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c14 {
  margin-left: 10px;
}

.c6 {
  padding: 9px;
  border-radius: 50%;
}

.c8 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.c7 {
  color: rgba(0,0,0,0.6);
}

.c7:hover {
  background-color: rgba(0,117,142,0.04);
}

.c7.Mui-checked,
.c7.MuiCheckbox-indeterminate {
  color: #00758e;
}

.c7.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
  margin: 0px;
}

.c4.Mui-disabled {
  cursor: default;
}

.c4 .MuiFormControlLabel-label.Mui-disabled {
  color: rgba(0,0,0,0.38);
}

.c11 {
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: 45px 1fr 78px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 12px 20px;
  background-color: #ffffff;
  box-shadow: 0px -1px 0px #dddddd;
  border-bottom: 1px solid #dddddd;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c13 {
  fill: #68b75f;
}

.c18 {
  fill: #00758e;
}

.c0 {
  width: 100%;
}

.c3 {
  width: 100%;
  padding: 12px 20px;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
}

.c2 {
  font-style: italic;
  font-size: 15px;
  line-height: 24px;
  margin-bottom: 5px;
}

@media print {
  .c5 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c7:hover {
    background-color: transparent;
  }
}

<body>
    <div>
      <div
        class="c0 MuiBox-root"
      >
        <p
          class="c1 MuiTypography-root MuiTypography-body1 c2"
        >
          Showing 1-1 of 1 results
        </p>
        <div
          class="c0 MuiBox-root"
        >
          <div
            class="c3 MuiBox-root"
          >
            <label
              class="c4 MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd"
            >
              <span
                class="c5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c6 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c7"
              >
                <input
                  class="c8 PrivateSwitchBase-input"
                  data-indeterminate="false"
                  type="checkbox"
                  value="SelectAll"
                />
                <svg
                  aria-hidden="true"
                  class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="c1 MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
              >
                Select all
              </span>
            </label>
          </div>
          <div
            class="c10 MuiBox-root"
          >
            <div
              class="c11 MuiBox-root"
            >
              <span
                class="c5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c6 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c7"
              >
                <input
                  class="c8 PrivateSwitchBase-input"
                  data-indeterminate="false"
                  type="checkbox"
                  value="mock-checked-value"
                />
                <svg
                  aria-hidden="true"
                  class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <div
                class="c12 MuiBox-root"
              >
                <svg
                  aria-hidden="true"
                  class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c13"
                  data-testid="AssignmentIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
                  />
                </svg>
                <div
                  class="c14 MuiBox-root"
                >
                  <p
                    class="c15 MuiTypography-root MuiTypography-body1"
                  >
                    mock-activity-name
                  </p>
                  <p
                    class="c16 MuiTypography-root MuiTypography-body1"
                  >
                    ID: mock-content-id Last Modified: Invalid Date
                  </p>
                </div>
              </div>
              <div
                class="c17 MuiBox-root"
                role="button"
              >
                <svg
                  aria-hidden="true"
                  class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c18"
                  data-testid="AddIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                <span
                  class="c19 MuiTypography-root MuiTypography-textSub"
                >
                  Add Resource
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c5::-moz-focus-inner {
  border-style: none;
}

.c5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
}

.c15 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #00758e;
  margin-bottom: 7px;
}

.c16 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  color: #383838;
}

.c19 {
  margin: 0;
  color: #080808;
  font-size: 13px;
  font-family: Source Sans Pro;
  font-weight: 400;
  font-style: normal;
  color: #383838;
  margin-top: 3px;
  line-height: 15px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c14 {
  margin-left: 10px;
}

.c6 {
  padding: 9px;
  border-radius: 50%;
}

.c8 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.c7 {
  color: rgba(0,0,0,0.6);
}

.c7:hover {
  background-color: rgba(0,117,142,0.04);
}

.c7.Mui-checked,
.c7.MuiCheckbox-indeterminate {
  color: #00758e;
}

.c7.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
  margin: 0px;
}

.c4.Mui-disabled {
  cursor: default;
}

.c4 .MuiFormControlLabel-label.Mui-disabled {
  color: rgba(0,0,0,0.38);
}

.c11 {
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: 45px 1fr 78px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 12px 20px;
  background-color: #ffffff;
  box-shadow: 0px -1px 0px #dddddd;
  border-bottom: 1px solid #dddddd;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c13 {
  fill: #68b75f;
}

.c18 {
  fill: #00758e;
}

.c0 {
  width: 100%;
}

.c3 {
  width: 100%;
  padding: 12px 20px;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
}

.c2 {
  font-style: italic;
  font-size: 15px;
  line-height: 24px;
  margin-bottom: 5px;
}

@media print {
  .c5 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c7:hover {
    background-color: transparent;
  }
}

<div>
    <div
      class="c0 MuiBox-root"
    >
      <p
        class="c1 MuiTypography-root MuiTypography-body1 c2"
      >
        Showing 1-1 of 1 results
      </p>
      <div
        class="c0 MuiBox-root"
      >
        <div
          class="c3 MuiBox-root"
        >
          <label
            class="c4 MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd"
          >
            <span
              class="c5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c6 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c7"
            >
              <input
                class="c8 PrivateSwitchBase-input"
                data-indeterminate="false"
                type="checkbox"
                value="SelectAll"
              />
              <svg
                aria-hidden="true"
                class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="c1 MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
            >
              Select all
            </span>
          </label>
        </div>
        <div
          class="c10 MuiBox-root"
        >
          <div
            class="c11 MuiBox-root"
          >
            <span
              class="c5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c6 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c7"
            >
              <input
                class="c8 PrivateSwitchBase-input"
                data-indeterminate="false"
                type="checkbox"
                value="mock-checked-value"
              />
              <svg
                aria-hidden="true"
                class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <div
              class="c12 MuiBox-root"
            >
              <svg
                aria-hidden="true"
                class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c13"
                data-testid="AssignmentIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
                />
              </svg>
              <div
                class="c14 MuiBox-root"
              >
                <p
                  class="c15 MuiTypography-root MuiTypography-body1"
                >
                  mock-activity-name
                </p>
                <p
                  class="c16 MuiTypography-root MuiTypography-body1"
                >
                  ID: mock-content-id Last Modified: Invalid Date
                </p>
              </div>
            </div>
            <div
              class="c17 MuiBox-root"
              role="button"
            >
              <svg
                aria-hidden="true"
                class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c18"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
              <span
                class="c19 MuiTypography-root MuiTypography-textSub"
              >
                Add Resource
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
